### ISSUE TITLE
SQL error during migration/update Formie 1.3.14 > 1.3.16.1

### ISSUE THREAD

**Issue by Timusan** [verbb/formie#335]
**Description**

```
Exit Code: 1(General error)

Output:
================
Yii Migration Tool (based on Yii v2.0.40)

Total 1 new Formie migration to be applied:
    - m210131_000000_form_data_retention

*** applying m210131_000000_form_data_retention
    > alter column dataRetention in table {{%formie_forms}} to string NOT NULL DEFAULT 'forever' CHECK ([[dataRetention]] in ('forever','minutes','hours','days','weeks','months','years')) ...Exception: SQLSTATE[42710]: Duplicate object: 7 ERROR:  constraint "formie_forms_dataretention_check" for relation "formie_forms" already exists
The SQL being executed was: ALTER TABLE "formie_forms" ALTER COLUMN "dataRetention" TYPE varchar(255), ALTER COLUMN "dataRetention" SET DEFAULT 'forever', ALTER COLUMN "dataRetention" SET NOT NULL, ADD CONSTRAINT formie_forms_dataRetention_check CHECK ("dataRetention" in ('forever','minutes','hours','days','weeks','months','years')) (/Users/tim/projects/wilms-web/vendor/yiisoft/yii2/db/Schema.php:678)
#0 /Users/tim/projects/wilms-web/vendor/yiisoft/yii2/db/Command.php(1304): yii\db\Schema->convertException(Object(PDOException), 'ALTER TABLE "fo...')
#1 /Users/tim/projects/wilms-web/vendor/yiisoft/yii2/db/Command.php(1099): yii\db\Command->internalExecute('ALTER TABLE "fo...')
#2 /Users/tim/projects/wilms-web/vendor/yiisoft/yii2/db/Migration.php(420): yii\db\Command->execute()
#3 /Users/tim/projects/wilms-web/vendor/verbb/formie/src/migrations/m210131_000000_form_data_retention.php(25): yii\db\Migration->alterColumn('{{%formie_forms...', 'dataRetention', Object(yii\db\ColumnSchemaBuilder))
#4 /Users/tim/projects/wilms-web/vendor/craftcms/cms/src/db/Migration.php(52): verbb\formie\migrations\m210131_000000_form_data_retention->safeUp()
#5 /Users/tim/projects/wilms-web/vendor/yiisoft/yii2/console/controllers/BaseMigrateController.php(744): craft\db\Migration->up()
#6 /Users/tim/projects/wilms-web/vendor/craftcms/cms/src/console/controllers/MigrateController.php(374): yii\console\controllers\BaseMigrateController->migrateUp('m210131_000000_...')
#7 [internal function]: craft\console\controllers\MigrateController->actionAll()
#8 /Users/tim/projects/wilms-web/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#9 /Users/tim/projects/wilms-web/vendor/yiisoft/yii2/base/Controller.php(181): yii\base\InlineAction->runWithParams(Array)
#10 /Users/tim/projects/wilms-web/vendor/yiisoft/yii2/console/Controller.php(184): yii\base\Controller->runAction('all', Array)
#11 /Users/tim/projects/wilms-web/vendor/yiisoft/yii2/base/Module.php(534): yii\console\Controller->runAction('all', Array)
#12 /Users/tim/projects/wilms-web/vendor/yiisoft/yii2/console/Application.php(181): yii\base\Module->runAction('migrate/all', Array)
#13 /Users/tim/projects/wilms-web/vendor/craftcms/cms/src/console/Application.php(89): yii\console\Application->runAction('migrate/all', Array)
#14 /Users/tim/projects/wilms-web/vendor/yiisoft/yii2/console/Application.php(148): craft\console\Application->runAction('migrate/all', Array)
#15 /Users/tim/projects/wilms-web/vendor/yiisoft/yii2/base/Application.php(392): yii\console\Application->handleRequest(Object(craft\console\Request))
#16 /Users/tim/projects/wilms-web/craft(23): yii\base\Application->run()
#17 {main}
*** failed to apply m210131_000000_form_data_retention (time: 0.041s)


0 from 1 migrations were applied.

Migration failed. The rest of the migrations are canceled.
```

**Steps to reproduce**
1. Have an installation on PostgreSQL of Formie 1.3.14.
2. Try to upgrade via console or web interface

**Additional info**
- Plugin version: 1.3.14
- Craft version: 3.6.1
- Multi-site: No

**Additional context**

Running on PostgreSQL.

**Comment by engram-design** [comment]
Should be fixed in the next release. To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.3.16.1",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Should be fixed in [1.3.17](https://github.com/verbb/formie/releases/tag/1.3.17)
